src/pdiagmat.jl

```diff
@@ -132,7 +127,7 @@ end
 ### tri products
 
 function X_A_Xt(a::PDiagMat, x::StridedMatrix)
-    z = x .* reshape(sqrt.(a.diag), 1, dim(a))
+    z = x .* sqrt.(reshape(a.diag, 1, dim(a)))
```

Comment:
This (and similar line below) should be fixed in any case. It seems the current implementation prevents broadcast fusion and creates an additional intermediate array.